packages/plugins/plugin-assistant/src/components/Timeline/Timeline.tsx

```diff
@@ -68,163 +27,278 @@ export type Commit = {
+            span.start = Math.min(span.start, getCommitIndex(parent));
+
+            // Detect merge.
+            if (parents.length > 1) {
+              const parentSpan = spans.get(branch);
+              if (parentSpan) {
+                parentSpan.end = Math.max(parentSpan.end, index);
+              }
             }
           }
         }
-      }
-    });
+      });
 
-    return spans;
-  }, [commits, branches]);
+      return spans;
+    }, [commits, branches]);
 
-  const getPaths = (commit: Commit, branch: string, span: Span, index: number): string[] => {
+    // Navigation.
+    const containerRef = useRef<HTMLDivElement>(null);
+    const [current, setCurrent] = useState<number | undefined>();
+    const currentCommit = useMemo(() => (current !== undefined ? commits[current] : undefined), [current, commits]);
+    useEffect(() => {
+      onCurrentChange?.({ current, commit: current === undefined ? undefined : commits[current] });
+      const el = containerRef.current?.querySelector(`[data-index="${current}"]`);
+      el?.scrollIntoView({ behavior: 'instant', block: 'nearest' });
+    }, [current]);
+    useEffect(() => {
+      return addEventListener(containerRef.current!, 'keydown', (event) => {
+        switch (event.key) {
+          case 'ArrowDown': {
+            event.preventDefault(); // Prevent implicit scrolling.
+            if (event.metaKey) {
+              setCurrent(commits.length - 1);
+            } else {
+              setCurrent((selected) => (selected === undefined ? 0 : Math.min(commits.length - 1, selected + 1)));
+            }
+            break;
+          }
+          case 'ArrowUp': {
+            event.preventDefault(); // Prevent implicit scrolling.
+            if (event.metaKey) {
+              setCurrent(0);
+            } else {
+              setCurrent((selected) => (selected === undefined ? commits.length - 1 : Math.max(0, selected - 1)));
+            }
+            break;
+          }
+          case 'Escape': {
+            setCurrent(undefined);
+            break;
+          }
+        }
+      });
+    }, [commits.length, containerRef.current]);
+
+    return (
+      <ScrollContainer ref={scrollerRef}>
+        <div tabIndex={0} className={mx('flex flex-col is-full !outline-none', classNames)} ref={containerRef}>
+          {commits.map((commit, index) => {
+            // Skip branches that are not whitelisted.
+            const idx = getBranchIndex(commit.branch);
+            if (idx === -1) {
+              return null;
+            }
+
+            return (
+              <div
+                key={commit.id}
+                data-index={index}
+                aria-current={current === index}
+                className={mx(
+                  'group flex shrink-0 items-center gap-2 overflow-hidden',
+                  // TODO(burdon): Factor out fragment?
+                  'aria-[current=true]:!bg-primary-500 hover:bg-hoverSurface',
```

Comment:
Please use semantic tokens, either an existing one or mint new ones, since we no longer guarantee physical tokens will be present.


```suggestion
                  'aria-[current=true]:!bg-accentSurface aria-[current=true]:hover:!bg-accentSurfaceHover hover:bg-hoverSurface',
```

Reply:
`aria-current` is a fairly niche attribute, I can only think of two places we use this, both master-detail patterns (navtree and mailbox), and both use the attribute to style items differently to suit the particular situation.

The pattern of applying `accentSurface` conditionally is also one I thought we intended to deprecate because it was too loud, except in cases where the component has attention. I can certainly imagine a fragment for that but this appears to be the first such case where we would do this.